types/abci.go

```diff
@@ -30,8 +30,13 @@ type ExtendVoteHandler func(Context, *abci.RequestExtendVote) (*abci.ResponseExt
 // pre-commit vote extension.
 type VerifyVoteExtensionHandler func(Context, *abci.RequestVerifyVoteExtension) (*abci.ResponseVerifyVoteExtension, error)
 
-// PreBlocker runs code before the `BeginBlocker`.
-type PreBlocker func(Context) (ResponsePreBlock, error)
+// PreBlocker runs code before the `BeginBlocker` and defines a function type alias for executing logic right
+// before FinalizeBlock is called (but after its context has been set up). It is
+// intended to allow applications to perform computation on vote extensions and
+// persist their results in state.
+//
+// Note: returning an error will make FinalizeBlock fail.
+type PreBlocker func(Context, *abci.RequestFinalizeBlock) (ResponsePreBlock, error)
```

Comment:
I wonder if we should be consistent with all ABCI methods & hooks where the response type is a pointer, i.e. `*ResponsePreBlock`.

Reply:
I don't have preference on it, but not sure we consider as API break change?

Reply:
It cannot be an API breaking change if we change it now ;)

Reply:
Do we need the separate api related change in another PR?

Reply:
No, doing everything in this PR is good. No changelog needed either as it never been released.